lib/Wikia/src/Service/Helios/HeliosClientImpl.php

```diff
@@ -206,39 +224,82 @@ public function invalidateToken( $token, $userId )
 	 *
 	 * @return array - JSON string deserialized into an associative array
 	 */
-	public function generateToken( $userId )
-	{
+	public function generateToken( $userId ) {
 		return $this->request(
-			sprintf('users/%s/tokens', $userId),
+			sprintf( 'users/%s/tokens', $userId ),
 			[],
 			[],
-			[ 'method' => 'POST' ]
+			[ self::METHOD => 'POST' ]
 		);
 	}
 
 	/**
-	* A shortcut method for register requests.
-	*/
-	public function register( $username, $password, $email, $birthdate, $langCode )
-	{
-			// Convert the array to URL-encoded query string, so the Content-Type
-			// for the POST request is application/x-www-form-urlencoded.
-			// It would be multipart/form-data which is not supported
-			// by the Helios service.
-			$postData = http_build_query( [
-				'username'  => $username,
-				'password'  => $password,
-				'email'     => $email,
-				'birthdate' => $birthdate,
-				'langCode'  => $langCode,
-			] );
-
-			return $this->request(
-				'users',
-				[],
-				$postData,
-				[ 'method'	=> 'POST' ]
-			);
+	 * A shortcut method for register requests.
+	 *
+	 * @param $username
+	 * @param $password
+	 * @param $email
+	 * @param $birthdate
+	 * @param $langCode
+	 *
+	 * @return mixed|null
+	 */
+	public function register( $username, $password, $email, $birthdate, $langCode ) {
+		// Convert the array to URL-encoded query string, so the Content-Type
+		// for the POST request is application/x-www-form-urlencoded.
+		// It would be multipart/form-data which is not supported
+		// by the Helios service.
+		$postData = [
+			self::USERNAME => $username,
+			self::PASSWORD => $password,
+			'email'        => $email,
+			'birthdate'    => $birthdate,
+			'langCode'     => $langCode,
+		];
+
+		return $this->request(
+			'users',
+			[],
+			$postData,
+			[ self::METHOD => 'POST' ]
+		);
+	}
+
+	public function setPassword( $userId, $password ) {
+		$postData = [
+			self::PASSWORD => $password,
+		];
+
+		return $this->request(
+			sprintf( 'users/%s/password', $userId ),
+			[],
+			$postData,
+			[ self::METHOD => 'PUT', self::HEADERS => [ Constants::HELIOS_AUTH_HEADER => $userId ] ]
+		);
 	}
 
+	public function validatePassword( $password, $name ) {
+		$postData = [
+			self::PASSWORD => $password,
+			self::USERNAME => $name,
+		];
+
+		return $this->request(
+			'password/validation',
+			[],
+			$postData,
+			[ self::METHOD => 'POST' ]
+		);
+	}
+
+	public function deletePassword( $userId ) {
+		return true;
+		// @todo Helios impl
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this "TODO" comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1135)